src/Formatter/Specialised/JoltHttpOGMTranslator.php

```diff
@@ -206,9 +208,18 @@ private function translateMap(stdClass $value): CypherMap
     {
         return new CypherMap(
             function () use ($value) {
-                /** @var stdClass|null $element */
+                /** @var stdClass|array|null $element */
                 foreach ((array) $value as $key => $element) {
-                    yield $key => $this->translateJoltType($element);
+                    // There is an odd case in the JOLT protocol when dealing with properties in a node. Lists appear not to receive a composite type label, which is why we have to handle them specifically here.
```

Comment:
I am slightly confused by the change you make here.
Because the $element comes from a `json_decoded(...,true)` string, the `array_key_first($element) === 0` check seems to be superfluous to me.

I removed the `array_key_first` check, but I am not sure what the use case/bug we're trying to circumvent is. Does this bug only give unexpected Lists, or can it also give unexpected Maps (Which would be a bit harder to prevent from throwing nasty bugs, because they are `stdClass` here).

Besides, I think this issue could use an upstream bug report in Neo4j?

Reply:
I made an [upstream bug report](https://github.com/neo4j/neo4j/issues/12858)

Reply:
Great work! Yes, the check was superfluous in hindsight